api/utils/requestProcessor.js

```diff
@@ -1613,18 +1613,26 @@ const processRequest = (params) => {
                                 params.qstring.query = {};
                             }
                         }
-                        if (params.qstring.query.$or) {
-                            params.qstring.query.$and = [
-                                {"$or": Object.assign([], params.qstring.query.$or) },
-                                {"$or": [{"global": {"$ne": false}}, {"creator": params.member._id + ""}]}
-                            ];
-                            delete params.qstring.query.$or;
+                        params.qstring.query.$and = [];
+                        if (params.qstring.query.creator && params.qstring.query.creator === params.member._id) {
+                            params.qstring.query.$and.push({"creator": params.member._id + ""});
                         }
                         else {
-                            params.qstring.query.$or = [{"global": {"$ne": false}}, {"creator": params.member._id + ""}];
+                            params.qstring.query.$and.push({"$or": [{"global": {"$ne": false}}, {"creator": params.member._id + ""}]});
+                        }
+                        if (params.qstring.app_id && params.qstring.app_id !== "") {
```

Comment:
We have to  restructurise it a bit.
(As explanation involves autenfigication - will leave comment in slack)